docs/usage.md

```diff
@@ -0,0 +1,151 @@
+  -h, --help                     display help for command
+```
+
+<details>
+  <summary>Click here to read more about supported values for the <code>&lt;template&gt;</code> parameter.</summary>
+  <br>
+  All [templates](template.md) are installable npm packages. Therefore, the value of <code>&lt;template&gt;</code> can be anything supported by <code>npm install</code>. Here's a summary of the possibilities:
+  <br><br>
+  <pre><code>
+  npm install [&lt;@scope&gt;/]&lt;name&gt;
+  npm install [&lt;@scope&gt;/]&lt;name&gt;@&lt;tag&gt;
+  npm install [&lt;@scope&gt;/]&lt;name&gt;@&lt;version&gt;
+  npm install [&lt;@scope&gt;/]&lt;name&gt;@&lt;version range&gt;
+  npm install &lt;git-host&gt;:&lt;git-user&gt;/&lt;repo-name&gt;
+  npm install &lt;git repo url&gt;
+  npm install &lt;tarball file&gt;
+  npm install &lt;tarball url&gt;
+  npm install &lt;folder&gt;</code></pre>
+</details>
+<br>
+
+### Global templates installed with yarn or npm
+
+You can preinstall templates globally before installing the Generator CLI. The generator first tries to locate the template in local dependencies; if absent it checks where the global generator packages are installed.
+
+```bash
+npm install -g @asyncapi/html-template@0.16.0
+ag asyncapi.yaml @asyncapi/html-template
+# The generator uses html-template version 0.16.0 and not the latest version.
+```
+
+### CLI usage examples
+
+**The shortest possible syntax:**
+```bash
+ag asyncapi.yaml @asyncapi/html-template
+```
+
+**Generating from a URL:**
+```bash
+ag https://bit.ly/asyncapi @asyncapi/html-template
+```
+
+**Specify where to put the result:**
+```bash
+ag asyncapi.yaml @asyncapi/html-template -o ./docs
+```
+
+**Passing parameters to templates:**
+```bash
+ag asyncapi.yaml @asyncapi/html-template -o ./docs -p title='Hello from param'
+```
+
+In the template you can use it like this: ` {{ params.title }}`
+
+**Disabling the hooks:**
+```bash
+ag asyncapi.yaml @asyncapi/html-template -o ./docs -d generate:before generate:after=foo,bar
+```
+
+The generator skips all hooks of the `generate:before` type and `foo`, `bar` hooks of the `generate:after` type.
+
+**Installing the template from a folder:**
+```bash
+ag asyncapi.yaml ~/my-template
+```
+
+It creates a symbolic link to the target directory (`~/my-template` in this case).
+
+**Installing the template from a git URL:**
+```bash
+ag asyncapi.yaml https://github.com/asyncapi/html-template.git
+```
+
+**Map schema references from baseUrl to local folder:**
+```bash
+ag test/docs/apiwithref.json @asyncapi/html-template -o ./build/ --force-write --map-base-url https://schema.example.com/crm/:./test/docs/
+```
+
+The parameter `--map-base-url` maps external schema references to local folders.
+
+### CLI usage with Docker
+
+We have a docker image that we regularly update and by using it you do not need to install Node.js or npm even though the generator is written with it since the Docker image has the Generator installed.
```

Comment:
```suggestion
When using our docker image that we regularly update, you do not need to install Node.js or npm, even though the Generator is written with it since the Docker image has the Generator installed.
```